libs/vkd3d-common/string.c

```diff
@@ -37,6 +37,19 @@ char *vkd3d_strdup(const char *str)
     return duped;
 }
 
+char *vkd3d_strdup_n(const char *str, size_t n)
+{
+    char *duped;
+    size_t len;
+
+    len = strnlen(str, n) + 1;
+
+    duped = vkd3d_malloc(len);
+    if (duped)
+        memcpy(duped, str, len);
```

Comment:
This doesn't look right, since it reads beyond n if incoming string is not null-terminated.

Reply:
It uses `strnlen` above which maxes out at the input `n`

Reply:
Right, but if input string is not null-terminated, won't strnlen just return n? `str[n]` would then be out of bounds.

Reply:
I'm dumb. Fixed locally.